lib/generators/sufia/templates/catalog_controller.rb

```diff
@@ -15,7 +15,7 @@ class CatalogController < ApplicationController
   # These before_filters apply the hydra access controls
   before_filter :enforce_show_permissions, only: :show
   # This applies appropriate access controls to all solr queries
-  CatalogController.search_params_logic += [:add_access_controls_to_solr_params, :add_advanced_parse_q_to_solr]
+  CatalogController.search_params_logic = [:add_advanced_parse_q_to_solr] + search_params_logic + [:add_access_controls_to_solr_params]
```

Comment:
What's the reason for putting the `add_access_controls_to_solr_params` at the end and the `add_advanced_parse_q_to_solr` at the beginning?